### ISSUE TITLE
OTU table failing vaildation

### ISSUE THREAD

**Issue by MissMcDolly** [joey711/phyloseq#1182]
Dear all,
I have an issue with importing my OTU table. Pyhloseq gives me the following error message: Error in .local(object, taxa_are_rows) : 
  argument "taxa_are_rows" is missing, with no default
As I'm a total newbie I don't know what is wrong, as I formatted my OTU table in excel according the Phyloseq tutorial https://vaulot.github.io/tutorials/Phyloseq_tutorial.html.
Attached my OTU table
[OTU_abundance_VD.xlsx](https://github.com/joey711/phyloseq/files/3359896/OTU_abundance_VD.xlsx)
Can you please help me either with a link to clear step-by-step tutorial or by posting and explaining what I do wrong?
Best wishes,
Veronika

**Comment by mikemc** [comment]
Please share the code you ran that gave the error message, so that we can see what the problem is. However, it sounds like you did not supply the `taxa_are_rows` argument when you tried to create the `otu_table` object. In that tutorial you linked to, note the line
```r
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
```
See also the help for `otu_table()` by typing `?otu_table` in the R terminal. It also sounds like you would benefit from looking at the phyloseq tutorials and vignettes (run browseVignettes("phyloseq") inside R).

**Comment by MissMcDolly** [comment]
So I managed the issue from yesterday, following your suggestion. 
I still have trouble integrating my data... now I'm facing the issue that sample names are different in the sample file and the OTU table.
I did the following:
 library(readxl)
> otu_mat<- read_excel("D:/Benutzer/mslv/Körnermaisstroh/Amplikonsequenzierung/16S/OTU_abundance_sorted.xlsx", 
+                                col_types = c("text", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric", "numeric", 
+                                              "numeric", "numeric"))
> tax_mat<- read_excel("D:/Benutzer/mslv/Körnermaisstroh/Amplikonsequenzierung/16S/OTU_taxonomy_sorted.xlsx")
> samples_df <- read_excel("D:/Benutzer/mslv/Körnermaisstroh/Amplikonsequenzierung/16S/metadata.xlsx")
> row.names(otu_mat) <- otu_mat$otu
Warning message:
Unknown or uninitialised column: 'otu'. 
> otu_mat <- otu_mat %>% select (-OTU)
> row.names(tax_mat) <- tax_mat$otu
Warning message:
Unknown or uninitialised column: 'otu'. 
> tax_mat <- tax_mat %>% select (-OTU)
> View(otu_mat)
> View(samples_df)
> View(tax_mat)
> otu_mat <- as.matrix(otu_mat)
> tax_mat <- as.matrix(tax_mat)
> OTU = otu_table(otu_mat, taxa_are_rows = TRUE)
> TAX = tax_table(tax_mat)
> samples = sample_data(samples_df)
> maizestraw<- phyloseq(OTU, TAX, samples)
Error in validObject(.Object) : invalid class “phyloseq” object: 
 Component sample names do not match.
 Try sample_names()
> maizestraw
Error: object 'maizestraw' not found
> sample_names(samples)
 [1] "sa1"  "sa2"  "sa3"  "sa4"  "sa5"  "sa6"  "sa7"  "sa8"  "sa9"  "sa10" "sa11" "sa12" "sa13" "sa14" "sa15" "sa16"
[17] "sa17" "sa18" "sa19" "sa20" "sa21" "sa22" "sa23" "sa24" "sa25" "sa26" "sa27" "sa28" "sa29" "sa30" "sa31" "sa32"
[33] "sa33" "sa34" "sa35" "sa36" "sa37" "sa38" "sa39" "sa40" "sa41" "sa42" "sa43" "sa44" "sa45" "sa46" "sa47" "sa48"
[49] "sa49" "sa50" "sa51" "sa52" "sa53" "sa54" "sa55" "sa56" "sa57" "sa58"

In my OTU table sample names are sample_names(OTU)
 [1] "G1"  "G2"  "G3"  "G4"  "G5"  "G6"  "G7"  "G8"  "G9"  "G10" "G11" "G12" "G13" "G14" "G15" "G16" "G17" "G18" "G19"
[20] "G20" "G21" "G22" "G23" "G24" "G25" "G26" "G27" "G28" "G29" "G30" "G31" "G32" "G33" "G34" "G35" "G36" "G37" "G38"
[39] "G39" "G40" "G41" "G42" "G43" "G44" "G45" "G46" "G47" "G48" "G49" "G50" "G51" "G52" "G53" "G54" "G55" "G56" "G57"
[58] "G58"
So I think I need to change the R given sampel names according the ones in the OTU table. Could I do that with sample_names<- ?
And thanks, I'm cross reading all the available tutorials, but as I'm programming for the first time, it is quiet hard to understand what is happening...
Thank you for your help!

**Comment by MissMcDolly** [comment]
Oh sorry, I forgot to attach the respective files.. now attached my OTU table and my metadata sheet
[OTU_abundance_sorted.xlsx](https://github.com/joey711/phyloseq/files/3363119/OTU_abundance_sorted.xlsx)
[metadata.xlsx](https://github.com/joey711/phyloseq/files/3363120/metadata.xlsx)

**Comment by mikemc** [comment]
The sample names are not getting set correctly when you create your sample_data object. You should set the rownames of the data frame to the sample names before calling `sample_data()`. The relevant lines in the tutorial you linked to are
```r
  row.names(samples_df) <- samples_df$sample
  samples_df <- samples_df %>% select (-sample) 
  samples = sample_data(samples_df)
```

**Comment by MissMcDolly** [comment]
Yes, I must have missed this! Thank you so much! Now I can finally try to continue!!!

**Comment by lisa-chandler** [comment]
Hi I am having an issue with sample names not getting set correctly.  I import my data from an excel file, and I have been using the following to create my metadata as a sample_data table
row.names(samples_df) <- samples_df$Sample
samples_df <- samples_df %>% select (-Sample)
samples = sample_data(samples_df)
But even though samples_df has what looks the correct row names, when I create the sample_data object it creates its own row names and I cannot transform into a phyloseq object.

I have used this line of code before with no issues, up until today so was just wondering there has been some sort of update in background that is causing this error. I did run an update for a few packages in RStudio late last week.
Thanks

**Comment by mikemc** [comment]
@lisa-chandler My guess would be that this has to do with a dplyr update. Are you sure that after the call to `select()` you still have a plain dataframe with rownames and not a tibble? You might have better luck using  `rownames_to_column()` from the tibble package which is meant for this purpose and should always give you a data frame with rownames
```r
samples_df <- samples_df %>%
  rownames_to_column("Sample")
samples = sample_data(samples_df)
```

**Comment by lisa-chandler** [comment]
Hi Mike thanks for the reply
not sure about plain dataframe? Am guessing I may have a tibble, as I have always got a warning message 'setting row names on a tibble is deprecated' but it has worked regardless. Will try your suggestion and see if that works

**Comment by lisa-chandler** [comment]
OK awesome - that worked! Thanks so much :)
